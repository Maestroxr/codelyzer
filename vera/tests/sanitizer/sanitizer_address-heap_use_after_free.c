#include <stdlib.h>
#include <stdio.h>

int main(int argc, char** argv) 
{
	int res = -1;
	int* array = (int*)malloc(100*sizeof(int));
	free(array);
	if (array[ argc ])// BOOM Heap-use-after-free	
	{
		res = 1;
	}
	return res;
}