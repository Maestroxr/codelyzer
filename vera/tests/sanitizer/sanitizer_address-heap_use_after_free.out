C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-heap_use_after_free.c:6: Pointer's 'array' malloc is missing null pointer check.
C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-heap_use_after_free.c:8: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_address-heap_use_after_free]: Heap use after free, suspect variables: ['array'], sanitizer_address-heap_use_after_free.c:8
	Was accessed by read of size 4 at:
		sanitizer_address-heap_use_after_free.c:8 in function 'main'
	Accessed address is located 4 bytes inside of 400-byte region, freed by thread T0 here:
		sanitizer_address-heap_use_after_free.c:7 in function 'main'
	Previously allocated by thread T0 here:
		sanitizer_address-heap_use_after_free.c:6 in function 'main'
:MemoryEnd:
